### PR TITLE
Handle coinor from brew as well

### PR DESCRIPTION
**Pull request by jpfeuffer**
## Description

makes OpenMS fully brew compatible.

We can actually simplify the mac docs with this.
Just add `brew install cbc` which has dependencies to all the coinor rest.

No contrib needed anymore.

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: cmake/Modules/FindCOIN.cmake [hunk 1]
```diff
@@ -143,7 +165,6 @@ find_package_handle_standard_args(COIN DEFAULT_MSG
 
 # export the libraries
 if(COIN_FOUND)
-  set(COIN_INCLUDE_DIRS ${COIN_INCLUDE_DIR})
```
Comment (timosachsenberg) [inline cmake/Modules/FindCOIN.cmake:146]: why can this be removed?

File: cmake/Modules/FindCOIN.cmake [hunk 2]
```diff
@@ -83,23 +86,44 @@ macro(_coin_find_lib _libname _lib_file_names _lib_file_names_debug)
...
+      list(APPEND INCLUDE_DIRS ${COIN_INCLUDE_DIR})
+    endif()
+
+    # Before, we handled the vcpkg specific sublibrary structure as follows. This should be handled equally
+    # by the more generic searching above (which also works for brew). But needs to be tested. I don't use
+    # vcpkg currently
+    # _libnamelower = FOLDER
+
+    #if(EXISTS "${COIN_INCLUDE_DIR}/${_libnamelower}/coin")
+    #  set(_libspecific_include "${COIN_INCLUDE_DIR}/${_libnamelower}" "${COIN_INCLUDE_DIR}/${_libnamelower}/coin")
```
Comment (timosachsenberg) [inline cmake/Modules/FindCOIN.cmake:113]: why did (and still do) both possibilities need to be added?

**Comment by timosachsenberg** [comment]
I can test vcpkg on linux

File: cmake/Modules/FindCOIN.cmake (hunk 1 above)
Comment (jpfeuffer) [reply cmake/Modules/FindCOIN.cmake:146]: we have a target based approach now. Actually we could add it back. I thought it caused an error. But none of those vars are used anymore.

File: cmake/Modules/FindCOIN.cmake (hunk 2 above)
Comment (jpfeuffer) [reply cmake/Modules/FindCOIN.cmake:113]: See the description above the lines.
We use `include "coin/CoinUtils.h"`
They use `include "CoinUtils.h"` for internal reference to other sublibs.

**Review by timosachsenberg — APPROVED** [review]

**Pull request merged**